src/sentry/services/hybrid_cloud/__init__.py

```diff
@@ -95,11 +97,9 @@ def close(self: Any) -> None:
 
     def make_method(method_name: str) -> Any:
         def method(self: Any, *args: Any, **kwds: Any) -> Any:
-            from django.test import override_settings
-
             from sentry.services.hybrid_cloud.auth import AuthenticationContext
 
-            with override_settings(SILO_MODE=target_mode):
+            with SiloMode.exit_single_process_silo_context():
```

Comment:
So this is another important change.  There are comments that explain this, but basically my reasoning was like this:

1.  We need thread safe context management to switch the thread between silo modes during acceptance tests.  
2. Previously we did so with `override_settings` but, tragically the django implementation is NOT thread safe.  Using `override_settings` causes all kinds of weird unexpected behaviors in acceptance tests as a result (weirdly, our actual servers in produciton use process pools for requests, but acceptance tests use threaded servers)
3. We also want to prevent unintended re-enter as that would be a bug.  Only re-entry that is the direct result of RPC should be allowed.